OAK-8138 | Switching statistical facet mode to secure in case hit cou…

nit0906 opened the pull request:
…nt is less than sample size

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- improve Collections.toArray using empty arrays

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848340 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- simplify expressions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848345 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- improvements to collection loops

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848385 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- use diamond type

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848394 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-301: Document Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848455 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-301: Document Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848487 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7943: Update Oak trunk to Jackrabbit 2.18.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848564 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.9.13 

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848585 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7948 - expose StatisticsProvider from QueryEngineSettings

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848592 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7954 - record index unavailable metric

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848593 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
apache jackrabbit Oak 1.9.13

fixing javadoc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848594 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.9.13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848597 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848599 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-7953: Test failure: JdbcToSegmentWithMetadataTest.validateMigration()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848615 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
adjust access were it can be weaker

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848662 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7956: Conflict may leave behind _collisions entry

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848729 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7956: Conflict may leave behind _collisions entry

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848769 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-7951: Datastore GC stats not updated with failure when "Not all repositories have marked references available"

Based on patch from Wim Symons
- Bubble up exception in case of not able to run sweep because of not all repositories not having references available
- Mark failure in stats which can then be queried

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848822 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-7951: Datastore GC stats not updated with failure when "Not all repositories have marked references available"

- Missing License header

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848823 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7959: MongoDocumentStore causes scan of entire nodes collection on startup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848837 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7961 - truncate up to first decimal place

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848861 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7962 - enable FV reranking by default

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848862 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848955 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- redundant checks for null
- access modifier can be weaker
- map.foreach instead of loop

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1848958 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- redundant array creation for vararg params

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849071 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- missing nullabiliy annotations in PrivilegeDdefinitionWriter
- more Map.forEach in PrivilegeDefinitionWriter
- unused securityProvider param in private method (UserImporter)
- redundant test for null in ImpersonationImpl

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849079 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- more access modifiers that must not be public
- PermissionValidator: redundant check for null node state
- PermissionStoreEditor: improve readability

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849091 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : Minor improvements to oak security code base
- fix indention in PermissionStoreEditor
- use Map.computeIfMissing in getMembership call, move getIdentifierManager and getPrincipalManager to inner classes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849104 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-7939: Create/Update documentation regarding secure facet counting

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849135 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7966 : Avoid adding excluded principal to cug policy

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849195 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7924 - added compressing codec

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849279 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-5918: Document enhancements in DocumentNodeStore in 1.6 - fix 'rep:documentStore' name in various places

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849308 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-5918: Document enhancements in DocumentNodeStore in 1.6 - make content structure examples code blocks to avoid rendering as lists

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849309 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7924 - removed unused import in compressing codec test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849364 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
apache jackrabbit Oak 1.8.10

baseline checks

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849384 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7947 Lazy loading of Lucene index files startup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1849465 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7947: Lazy loading of Lucene index files startup - reverted change due to test failures

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850163 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7969: Update tika dependency to 1.20

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850221 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7947 Lazy loading of Lucene index files startup (v4)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850229 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7947 Lazy loading of Lucene index files startup (related changes)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850231 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-7719 - Let check use the specified journal consistently

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850238 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7970: RDB*Store: add profile for DB2 11.1 JDBC driver

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850319 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7971: RDB*Store: update DB2 JDBC reference to 4.19.77

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850343 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7968 Active deletion of Lucene binaries: configuration option to disable

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850741 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7944 : revert changes made to SecurityProviderImpl in rev. 1848662 (public access modifier on constructor)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850752 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7947 Lazy loading of Lucene index files startup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850826 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-7975: Facet extraction fails while requesting multiple facets and one of the requested facets doesn't have indexed values

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850837 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850846 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare branch 1.10

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850861 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850863 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.10.0

fixing missing versions after branching



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850867 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7978: guava-latest profile defunct -> fixed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850874 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.10.0

fixing new SNAPSHOT version for trunk. it was wrongly 1.11-SNAPSHOT
now is correctly 1.12-SNAPSHOT



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850878 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7979: DeclaredMembershipPredicate does not compile with Guava 20

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1850882 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7983, OAK-7947 - temporarily reverting lazy loading patch, until OAK-7983 gets fixed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851022 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7984: Batch update documents in commit rollback

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851236 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7986: Incorrect MongoDB URI in ReplicaSetDefaultWriteConcernIT

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851246 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7984: Batch update documents in commit rollback

Use batch DocumentStore.createOrUpdate() and enable test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851253 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7976: Non-blocking commit rollback

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851330 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7976: Non-blocking commit rollback

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851334 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7974: Detailed merge metrics

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851392 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-7987: LdapIdentityProviderImpl#getIdentity(ExternalIdentityRef) won't work with useUidForExtId enabled

LdapIdentityProviderImpl#getIdentity() now searches correctly for uids/names.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851403 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7974: Detailed merge metrics

Document new metrics

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851419 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.10.0

baseline checks

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851442 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7982 : ACL.addEntry: check for mandatory restrictions only respects single value restrictions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851451 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7988 The node counter jmx bean should show 0 if a node exists

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851453 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-7903 - introduced constant for metric name

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851505 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
new release schedule

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851512 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-6749 - Extract the blob processing logic from StandbyDiff

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851533 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-6749 - Test handling of in-memory blob IDs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851534 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-6749 - Add RemoteBlobProcessor

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851535 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-6749 - Test that inline binaries are never downloaded

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851551 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-301: Document Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851605 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-6749 - Don't trigger a binary download request for in-memory Blob IDs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851619 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7993 : CompositeAuthorizationConfiguration.getRestrictionProvider() should filter duplications

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851625 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-7977 - Add multi-threaded segment transfer to oak-run segment-copy

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1851789 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7947 Lazy loading of Lucene index files startup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852007 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8003: MongoDocumentStore does not log server details

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852052 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8001: Lucene index can be empty (no :data node) in composite node store setup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852084 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7904 Exporting query duration per index metrics with Sling Metrics / DropWizard

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852102 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8004: oak-run: support "recovery" command for RDBDocumentStore

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852120 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8002: RDBDocumentStore: add RDB-specific MissingLastRevSeeker

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852135 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-7977 - Add multi-threaded segment transfer to oak-run segment-copy
Simplified segments migration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852357 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-8006 - SegmentBlob#readLongBlobId might cause SegmentNotFoundException on standby

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852424 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8009: Remove unused code in NodeDocument

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852449 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8007: RDBDocumentStore: potential off-heap memory leakage due to unclosed GzipInputStream

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852451 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8008: Unmerged branch changes visible after restart

Add an ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852460 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8008: Unmerged branch changes visible after restart

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852461 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8008: Unmerged branch changes visible after restart

Revert changes from r1852460 and r1852461

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852491 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8012: Unmerged branch changes visible after restart

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852492 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8012: Unmerged branch changes visible after restart

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852493 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7960: RDB: add to Oak documentationRDB: add to Oak documentation (work in progress)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852528 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7960: add link targets for Oak documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852582 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7960: RDB: add to Oak documentationRDB: add to Oak documentation (work in progress)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852584 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8016: RDBDocumentStore: minor improvements to GZIP compression of BLOB contents

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852601 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-8006 - SegmentBlob#readLongBlobId might cause SegmentNotFoundException on standby
Added test case
Introduced eagerSegmentCaching option for the FileStore to allow caching segments right after creating them

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852612 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
minor improvement: adjust svn:ignore

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852683 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8018 : Move LazyValue from oak-core to oak-commons

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852689 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8019 : Replace deprecated NodeUtil in AbstractOakCoreTest

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852696 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8020 : Create ImmutableACL from another AbstractAccessControlList

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852728 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
minor improvement: missing @Nullable with exercise principalprovider

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852741 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7997 : Adding restrictions to ACLs yields empty results for queries in Jackrabbit Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852758 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8021 : PrivilegeBitsProvider.getBits(Privilege[],NameMapper) should use getOakNameOrNull 

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852763 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852895 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8022: Update Oak trunk to Jackrabbit 2.19.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852907 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8017: Test failure: LastRevRecoveryRandomizedIT

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1852920 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-4521 Oak-run: add profiling and analysis tools (support for module names; sort files by name)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853009 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8025: Improve branch state comparison

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853054 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8026: Warn message when branch is created

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853055 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8028: Avoid unnecessary query in UnsavedModifications.persist()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853077 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8024 : OAK-8024 (patch provided by Ruben Reusser)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853083 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-7997 : Adding restrictions to ACLs yields empty results for queries in Jackrabbit Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853085 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-3574: Query engine: support p=lowercase('x') and other function-based indexes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853119 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8029: Additional merge metrics

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853131 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8029: Additional merge metrics

Add documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853132 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8030: oak-jcr NodeTypeTest improvements

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853141 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8031: Update Oak trunk to Jackrabbit 2.19.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853147 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-7997 : Adding restrictions to ACLs yields empty results for queries in Jackrabbit Oak (rename and simplify a bit)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853206 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8036 : Improve test in oak-security-spi

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853226 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-6221 Deprecate SecurityProviderImpl
 - trivial javadoc note



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853228 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8037: add test case for making a node type referenceable

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853229 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8039: Intermittent test failure of CompactionAndCleanupIT.testMixedSegments
Report all exceptions from background threads

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853331 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8039: Intermittent test failure of CompactionAndCleanupIT.testMixedSegments
Add assertion to root cause of OAK-8014 to determine whether that issue also causes this issue

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853332 13f79535-47bb-0310-9956-ffa450edef68

Commit:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853355 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8039: Intermittent test failure of CompactionAndCleanupIT.testMixedSegments
Include stack traces of exceptions in background thread

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853379 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8039: Intermittent test failure of CompactionAndCleanupIT.testMixedSegments
Better error message in assertion

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853389 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-7960: RDB: add to Oak documentation - tune RDBHelper output

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853393 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8027 : Extract public utility for handling jcr:all privilege bits

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853421 13f79535-47bb-0310-9956-ffa450edef68

Commit:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853427 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8033: Node states sometimes refer to more than a single generation of segments after a full compaction
Use the gc-generation of the current write operation instead of the one of the repository head state when allocating new segment buffer writers in the course of a write operation.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853429 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8043: RDB: expose DDL generation functionality in oak-run

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853433 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8042: IndexDefinitnionBuider should support deprecated properties on index definition

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853434 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8041: IndexDefinitnionBuider should support facets and boost for property definitions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853435 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8036 : Improve test in oak-security-spi
minor improvement: missing @NotNull/@Nullable  annotations with ImmutablePrivilegeDefinition constructor

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853436 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8023 : AccessControlManagerImpl can not handle repository level when editing policies by principal

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853441 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8044 : AccessControlManagerImpl.getEffectivePolicies returns empty ACLs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853449 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8000 : AccessControlManagerImpl.getEffectivePolicies(String) doesn't respect restrictions (tests and docu update)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853451 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8035 - a message is logged when similar costs come from the same index type

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853626 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8050 - set CompressingCodec via system property

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853629 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853632 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8049 : AbstractSecurityTest: fix PartialValueFactory field and it's usages

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853645 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8035: a message is logged when similar costs come from the same index type

fix test memory leak due to log appending, fix by Nitin Gupta

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853659 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8043: RDB: expose DDL generation functionality in oak-run - update oak-doc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853808 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8060: Incorrect read preference when parentId refers to NodeDocument.NULL

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853861 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8060: Incorrect read preference when parentId refers to NodeDocument.NULL

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853862 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8058: RDB*Store: update Tomcat JDBC pool dependency to 8.5.38

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853866 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8054 RepMembersConflictHandler creates property with wrong type



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853868 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8059: Update Jackson dependency to 2.9.8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853870 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8035: a message is logged when similar costs come from the same index type

test cleanup, use LogCustomizer, remove assertj dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853872 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8043: RDB: expose DDL generation functionality in oak-run - update oak-doc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853882 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7960: RDB: add to Oak documentation - link to oak-run rdbddldump description

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853893 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8046: Result items are not always correctly counted against the configured read limit if a query uses a lucene index

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853969 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8046: Result items are not always correctly counted against the configured read limit if a query uses a lucene index

Fix test cases. I had earlier assumed that index tag chooses even
aggregate index (compatVersion=1) but it doesn't. So, now we have
different type of indexes and query according to which one we want to
get picked.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1853997 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8065 : AbstractSecurityTest.getAccessControlManager: should use getNamePathMapper() instead of DEFAULT

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854017 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8052: PersistentCache: failure during construction may lead to resource leak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854034 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8070: The date-based copy-versions directive doesn't work correctly with include-paths

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854038 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8068: Update slf4j dependency to 1.7.26

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854044 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8072 - aggregate jcr:content search results as their parents

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854055 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8067 Measure fsync (called when closing the NRT index)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854057 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8069: Log warning for too many transient modifications of direct child nodes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854058 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8074: RDB*Store: update mysql-connector-java dependency to 8.0.15

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854113 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8067 Measure fsync (called when closing the NRT index) and try to reduce disk I/O

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854115 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-7994 Principal Management APIs don't allow for search pagination



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854314 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8053 Add intermediate report to ExternalLoginTest



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854316 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - streamline Javadoc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854373 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - hints for MySQL

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854377 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - hints for DB2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854380 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - remove DB2 and MySQL specific hints that we moved to oak-doc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854385 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - hints for PostgreSQL

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854401 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8080: RDB*Store: move DB-specific config hints from Javadoc into oak-doc - remove PostgreSQL specific hints that we moved to oak-doc

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854403 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8051: PersistentCache: error during open can lead to incomplete initialization and subsequent NPEs - cleanup trailing whitespace in source code

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854455 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8084: LogCustomizer should allow instantiation with Java class (in addition to class name)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854461 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8085: Upgrade spotbugs to 3.1.11

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854462 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8088: Add refresh head revision time to background update stats

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854466 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8087: RDB*Store: update mssql-jdbc driver reference to 7.2.1.jre8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854468 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8013: Disable filename* part of Content-Disposition specification for direct binary access.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854489 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8090: Update baseline comparisonVersion to 1.10.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854499 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8071: Logging to detect commits carrying over from previous GC generation can block other threads from committing
Log warnings for commits being blocked on the current commit taking a long time or the current commit crossing a gc boundary.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854515 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8062 : PrincipalProvider: optional lookup of ItemBasedPrincipal by path

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854530 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8095 VersionStorageEditor can prevent upgrade of version store



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854533 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8096 : CompositeAccessControlManager: misleading local field name

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854535 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8098: oak-examples/webapp: update spring boot dependency to 1.5.19

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854539 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8097 Load Lucene index files before writing to the index

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854565 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8066: Nodes with many direct children can lead to OOME when saving
Test case

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854579 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8066: Nodes with many direct children can lead to OOME when saving
Removed left over comment

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854580 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8066: Nodes with many direct children can lead to OOME when saving
Flush modified child nodes to disk in regular intervals

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854596 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854681 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8051: PersistentCache: error during open can lead to incomplete initialization and subsequent NPEs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854701 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8093 : AccessControlManagerImpl: respect local namespaces when writing back ACEs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854756 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854758 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-8063 - The cold standby client doesn't correctly handle backward references

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854772 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8089: DocumentNodeStore dispose can fail when duration of final background ops exceeds lease time

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854773 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8083: RDB*Store: add SQLServer specific documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854774 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8106: High memory usage when large branch is reset

Add an ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854827 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8108: Branch reset does not remove all branch commit entries

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854848 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8097 Load Lucene index files before writing to the index (test)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854855 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8108: Branch reset does not remove all branch commit entries

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854859 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8101 : AccessControlValidator prevents alternative authorization models to use restrictions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854921 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8106: High memory usage when large branch is reset

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854930 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8102 LoginModule error metrics



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854982 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8181 - added optional 'tag' clauses for similarity search

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854990 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8119 - use exact fv distance score for similarity search

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1854991 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8120: Update http components to 4.5.7/4.4.11

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855032 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-7027 - Correctly test the read timeout expiration on cold standby

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855048 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8121: RDBExport should handle db2 dump where del files refers to lob files without offset and length params

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855130 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8122: MongoDocumentStore init may fail with authentication

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855221 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8103: TimingHookTest.commitTime failing occasionally

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855228 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8078 UserPrincipalProvider support for range search



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855333 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8125 PrincipalProviderImpl support for range search



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855340 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8127 : TreeLocation: imcomplete Nullable/NotNull annotations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855401 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8117 : NPE when adding ACE with restrictions and remapped namespaces

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855402 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8112: Change log pattern to have complete date for oak-run datastore command

Changed pattern to have the complete date

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855477 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8115: Allow disabling of BlobTracker when not required

- BlobTracker can be disabled by setting config property snapshotIntervalSecs to 0 or below as this prop needs a sensible non-zero value anyways to be effective

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855478 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8130 - update corruption stats also when no failing indexes remain

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855496 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8114: IndexDefinitionBuilder should be smarter when to reindex while updating a definition

Applying work done by Nitin Gupta at https://github.com/apache/jackrabbit-oak/pull/122. Thanks Nitin for the contribution.

Did a few more changes on top:
* check refresh only when reindex isn't required
* we had forgotten that index tags also don't require reindexing

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855522 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8131 Principal Management APIs full text support



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855533 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8131 Principal Management APIs full text support



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855536 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8124: Sidegrade operation doesn't run security-related commit hooks

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855570 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8094: JMX monitoring to detect commits carrying over from previous GC generation can block other threads from committing
Always handle commits as queued even when the lock is free to avoid races between checking the number of available permits and actually acquiring the lock

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855586 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8094: JMX monitoring to detect commits carrying over from previous GC generation can block other threads from committing
Simplify CommitsTrackerTest

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855587 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8094: JMX monitoring to detect commits carrying over from previous GC generation can block other threads from committing
Add timestamp to monitoring of queued commits

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855588 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8094: JMX monitoring to detect commits carrying over from previous GC generation can block other threads from committing
Track and expose commit in progress and queued, dequed and applied time stamps per commit

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855589 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-8094: JMX monitoring to detect commits carrying over from previous GC generation can block other threads from committing
Expose gc generation of commit

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855590 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8137: Oak run tooling to run a RepositoryInitializer on top a list index definitions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855723 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8140 UserPrincipalProvider support for full text search




git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855771 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8111: Create read-only DocumentNodeStore for oak-run recovery dry run

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855776 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8138 | Switching statistical facet mode to secure in case hit count is less than sample size

catholicon commented on oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java:
```diff
@@ -365,7 +365,7 @@ private boolean loadDocs() {
                             if (facetProvider == null) {
                                 facetProvider = new LuceneFacetProvider(
-                                        FacetHelper.getFacets(searcher, query, plan, indexNode.getDefinition().getSecureFacetConfiguration())
+                                        FacetHelper.getFacets(searcher, query, plan, indexNode.getDefinition().getSecureFacetConfiguration(), docs.totalHits)
```
This `docs` object is first batch of results that fills up cursor. While `totalHits` might be accurate (it most likely is but I'd need to double check it), but I would prefer if we don't need to pass this on most likely because this requires an external agent to pass on an information that is actually implementation requirement and the external agent has a potential to mess up some internal assumption.

I understand with current code structure (switch deciding which type of facet counter to use quite early) it's easier to pass on this information. But, maybe, what we can do is to let `StatisticalSortedSetDocValuesFacetCounts#getTopChildren0` start doing its work and if it realizes that hit count is small enough then it can delegate to a new instance of `SortedSetDocValuesFacetCounts`. Wdyt?

nit0906 replied:
totalHits was accurate - that I checked - for instance - in case the query had 400 valid results but docs had only 50 top docs - then docs.size was equal to 50 but docs.totalHits was 400 ..I validated this with a couple of different use cases.

But yes - I agree with you , that it might be better if we don't pass total hits here (even I was a bit skeptical about this and wanted to get the approach reviewed). Let me check if we can work on a different approach as you suggested .

catholicon commented on oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/util/FacetHelper.java:
```diff
@@ -62,6 +62,14 @@ public static FacetsConfig getFacetsConfig(NodeBuilder definition) {
 
     public static Facets getFacets(IndexSearcher searcher, Query query, QueryIndex.IndexPlan plan,
                                    SecureFacetConfiguration secureFacetConfiguration) throws IOException {
+        // This method is just a fail safe in case it is being called via the old signature in any code that consumes oak.
```
I don't think we need to worry about these - this class should not be exposed outside oak-lucene and it should be ok to change method signature any way we want.

catholicon commented on oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/SecureFacetTest.java:
```diff
@@ -172,6 +172,51 @@ private void createLargeDataset() throws RepositoryException {
         assertNotEquals("Acl-ed and actual counts mustn't be same", actualLabelCount, actualAclLabelCount);
     }
 
+    private void createSmallDataset() throws RepositoryException{
```
I haven't read the test, but please do a quick check of existing tests that they are working with "larger than threshold" data set...

nit0906 replied:
Ran all the tests - they were all green .

catholicon replied:
no, i didn't mean if they pass or not... with this issue fixed, lower result size would still give "right" results expected from a statistical pov... I think we should make sure that existing tests create/get-results which are bigger that sample size threshold.

catholicon replied:
ok... ignore the noise I created... just read the tests again and found that `createLargeDataset` already is creating num_labels * sample_size nodes (and including all of them in results)... I just wasn't sure if that was ensure already or not.

stillalex committed:
OAK-8142 CompositePrincipalProvider support for full text search



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855838 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8113: Add ability to push metrics to prometheus pushgateway from oak-run

- Added an option --export-metrics to enable export
- MetricsExporterFixture and FixtureProvider initialize metrics exporter based on type (currently only pushgateway).
- Additional io.prometheus.simpleclient* libraries made optional and not part of the jar and would need to be specified externally on the classpath

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855842 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8143: Configurable persistent cache for Oak-Mongo fixture

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855846 13f79535-47bb-0310-9956-ffa450edef68

catholicon commented on oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/SecureFacetTest.java:
```diff
@@ -172,6 +172,51 @@ private void createLargeDataset() throws RepositoryException {
+
+        for (int i = 0; i < 4; i++) {
+            Node subPar = par.addNode("par" + i);
+            for (int j = 0; j < NUM_LEAF_NODES/10; j++) {
```
```suggestion
            for (int j = 0; j < NUM_LEAF_NODES/(2 * NUM_LABELS); j++) {
```

just to kind of keep num labels fixed... btw, I think we should also replace `4` with `NUM_LABELS` both here in and in `createLargeDataset`

reschke committed:
OAK-8150: RDB*Store: add Oracle specific documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855905 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8126 Load Lucene index files before writing to the index: close directory

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855906 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
GRANITE-24239 Composite node store: tests with queries

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855907 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855980 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8151 : Let ACE.getPrincipal return principals obtained from PrincipalManager

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855986 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8135: HTTP service may not select correct media type if multiple are specified in Accept header field

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855993 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8156: Update baseline comparisonVersion to 1.10.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1855996 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7991 Composite node store: tests with queries (code documentation)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856033 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-301: Document Oak
Memoirs in Garbage Collection (WIP)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856038 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856041 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8147: RDBBlobStore: add perf logging for JDBC read operations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856049 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8159 : GroupPrincipalWrapper.getName is redundant

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856054 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8163: examples: update Tomcat dependency to 7.0.93

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856056 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8160 : Missing notnull/nullable annotations with GroupPrincipals

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856059 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8164 : CompositePrincipalProvider.findPrincipals lacks NotNull and Overrides annotation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856060 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8036 : Improve test in oak-security-spi (wip)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856068 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-301: Document Oak
Memoirs in Garbage Collection (WIP) -- fixed header for new file



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856069 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8138 | Incorporating Vikas's review comments - moving the logic to compare hitCount and SampleSize to getTopChildren0 in StatisticalSortedDocValueFAcetCounts

Commit:
OAK-8138 | Trivial formatting - removing unintented line break

Commit:
Merge branch 'upstream-trunk' into OAK-8138
